### PR TITLE
Upgrade base container to PyTorch 23.02

### PR DESCRIPTION
**Pull request by ericharper**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 pipeline {
   agent {
         docker {
-          image 'nemo_containers:23.01_apex_c3d575f2478cd379b3c2d81f41edde39791b5d92'
+          image 'nvcr.io/nvidia/pytorch:23.02-py3'
```
Comment (XuesongYang) [inline Jenkinsfile:4]: is it on purpose to switch to a Pytorch container from the nemo container?

Comment (XuesongYang) [reply Jenkinsfile:4]: ah I see. Thank your for explanation.

Comment (ericharper) [reply Jenkinsfile]: Yes I'm sorry for this confusion. But we're only taking the PyTorch base container and installing different version of apex. It's not actually a nemo container, I will change the name

**Review by titu1994 — APPROVED** [review]
Looks ok for now, will we patch the 2 tests during the bugfix cycle ?

**Pull request merged**